models/world_coordinates.py

```diff
@@ -152,4 +120,72 @@ def get_downstreams(self, session=None):
 
         downstreams = zps + subs
         return downstreams
+
+    def save( self, filename=None, **kwargs ):
+        """Write the WCS data to disk.
+        Updates self.filepath
+        Parameters
+        ----------
+          filename: str or path
+             The path to the file to write, relative to the local store
+             root.  Do not include the extension (e.g. '.psf') at the
+             end of the name; that will be added automatically.
+             If None, will call image.invent_filepath() to get a
+             filestore-standard filename and directory.
+          Additional arguments are passed on to FileOnDiskMixin.save
+        """ 
+
+        # ----- Make sure we have a path ----- #
+        # if filename already exists, check it is correct and use
+
+        if filename is not None:
+            if not filename.endswith('.txt'):
+                filename += '.txt'
+            self.filepath = filename
+
+        # if not, generate one
+        else:
+            if self.image.filepath is not None:
+                self.filepath = self.image.filepath
+            else:
+                self.filepath = self.image.invent_filepath()
+
+            if self.provenance is None:
+                raise RuntimeError("Can't invent a filepath for the WCS without a provenance")
+            self.filepath += f'.wcs_{self.provenance.id[:6]}.txt'
+
+        txtpath = pathlib.Path( self.local_path ) / f'{self.filepath}'
+
+        # ----- Get the header string to save and save ----- #
+        header_txt = self.wcs.to_header().tostring(padding=False, sep='\\n' )
+
+        if txtpath.exists():
+            if ( 'overwrite' in kwargs and not kwargs['overwrite'] ):
+                # raise the error if overwrite is explicitly set False
+                raise FileExistsError( f"{txtpath} already exists, cannot save." )
+
+        with open( txtpath, "w") as ofp:
+            ofp.write( header_txt )
+
+        # ----- Write to the archive ----- #
+        FileOnDiskMixin.save( self, txtpath, **kwargs )
+
+    def load( self, download=True, always_verify_md5=False, txtpath=None ):
+        """Load this wcs from the file.
+        updates self.wcs.
+        Parameters
+        ----------
+        txtpath: str, Path, or None
+            File to read. If None, will load the file returned by self.get_fullpath()
+        """
+
+        if txtpath is None:
+            txtpath = self.get_fullpath( download=download, always_verify_md5=always_verify_md5)
+
+        if txtpath is None:
+            raise ValueError("WCS object has no filepath locally or in archive.")
```

Comment:
This error might be a little misleading. The filepath doesn't come from the archive, it is either defined on the object (i.e., on the database row of this WorldCoordinates) or it is given by the user as an override (this will usually not be the case). 

You can say we didn't find a file with that filepath not locally or on the archive, but that would be an error raised by `get_fullpath` and not here.

Reply:
You could add a check to see if the file exists (e.g., `os.path.isfile(txtpath)`) and then say something like "file is missing". This could happen if you specify `download=False` and it doesn't exist locally (even when the filepath exists).

Reply:
Would you say I should just remove the error you commented about then, and leave it to be caught somewhere in `get_fullpath` then?

I added the check for file existence and a test for it in `test_save_and_load`.